### DISCUSSION TITLE
No cron jobs for CMA, JMA, BOM

### DISCUSSION THREAD

**Discussion by drmrbrewer** [open-meteo/open-meteo#752] [General]
The output from `openmeteo-api cronjob` provides list of suitable entries for a cron file for regular updating of the relevant data.  I assume this comes [from here](https://github.com/open-meteo/open-meteo/blob/main/Sources/App/Commands/CronjobCommand.swift).  

But I can't see any cron jobs corresponding to `download-cma`, `download-jma` or `download-bom`.  Why is that?

**Comment by patrick-zippenfenig** [comment]
All three do not provide access to the general public, although all data is open-data. Similar, ERA5 or CERRA requires a CDS account. Another example is GloFAS flood data and global air quality data. Both require access to an ECMWF server with additional credentials. Please note that, open-data servers from national server services have limited bandwidth and it is very costly to operate open-data servers at this scale. Many open-data GRIB file distributions do offer the option to select individual weather variables and users have to download global GRIB files with all weather variables. I do not want to stretch their systems too far. 

As an alternative you can get the converted open-meteo database files directly from the [open-data AWS distribution](https://github.com/open-meteo/open-data). As weather variables can be selected individually, it is more space/bandwidth efficient and by using S3 it can scale better.

**Reply by drmrbrewer** [reply]
So what is e.g. `openmeteo-api download-cma` actually meant to do?

**Reply by patrick-zippenfenig** [reply]
They are used on Open-Meteo servers to download GRIB files, convert them to the open-meteo database format and publish data back to AWS. They accept a `--server` argument to specify a http server or proxy with username password credentials like `https://username:password@myserver.tld/data/YYYY-MM-DD/HH/`.

**Reply by drmrbrewer** [reply]
Ah OK, so not intended for "general use" for the reasons given above (additional credentials required)... perhaps a brief explanation about this could be provided in the help info (i.e. when running `openmeteo-api download-cma -h`)?
